python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -188,10 +190,27 @@ def add_input(self, refs: RefBundle, input_index: int):
             bundle = self._block_ref_bundler.get_next_bundle()
             self._add_bundled_input(bundle)
 
-    def _get_runtime_ray_remote_args(self) -> Dict[str, Any]:
+    def _get_runtime_ray_remote_args(
+        self, input_bundle: Optional[RefBundle] = None
+    ) -> Dict[str, Any]:
+        ray_remote_args = copy.deepcopy(self._ray_remote_args)
+        if "scheduling_strategy" not in ray_remote_args:
+            ctx = DataContext.get_current()
+            if input_bundle and input_bundle.size_bytes() > ctx.large_args_threshold:
+                ray_remote_args[
+                    "scheduling_strategy"
+                ] = ctx.scheduling_strategy_large_args
+                # Takes precedence over small args case. This is to let users know
+                # when the large args case is being triggered.
```

Comment:
We can revisit, but if these are user-facing, I wonder if we should make the tag more descriptive? Like `scheduling_strategy: LOCALITY-AWARE` or something.

Reply:
I agree, we could add LOCALITY_AWARE and make DEFAULT an alias in core. cc @jjyao